mavros/src/plugins/imu_pub.cpp

```diff
@@ -207,13 +206,12 @@ class IMUPubPlugin : public MavRosPlugin {
 
 		auto imu_msg = boost::make_shared<sensor_msgs::Imu>();
 
-		// PX4 NED (w x y z) -> ROS ENU (x -y -z w) (body-fixed)
-		tf::Quaternion orientation(att_q.q2, -att_q.q3, -att_q.q4, att_q.q1);
+		tf::Quaternion orientation(att_q.q1, att_q.q2, att_q.q3, att_q.q4);
```

Comment:
NO! You missed that FCU and ROS use different field convention.
It still should be `orientation(q2, q3, q4, q1)`.